bigquery/query_builder.py

```diff
@@ -397,3 +424,7 @@ def _render_limit(limit):
         return ''
 
     return "LIMIT %s" % limit
+
+
+def _replace_quote_with(match):
+    return match.group(0)[0] + r"\'"
```

Comment:
👀 this is a little bit hard to get it

Reply:
That's why it is wrapped in function with a descriptive name: the function replaces a match with te string returned by it

Reply:
Yes, the name of the function is well descriptive. I said that because of the param and the body... 

Is `match` a collection or a string or a regex or what is? 
What does `group(0)[0]` do here? 
Why `(0)[0]`? 
What does `r"\'"` means?. 

You know, the code is not so explicit (not because you, but because the way these regex functions are implemented in the language). If a new developer has to modify this function, at least he/she should has a good understanding of regex API, to make sure everything continue working as desired after modification.

Reply:
Got it, I will add a comment.

Reply:
Perhaps you can also make this regex a constant, like you do above with UNESCAPED_QUOTE, just to make things a little easier to read.